plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -45,13 +45,18 @@ import           Control.Monad
 import           Control.Monad.Reader
 import           PlutusPrelude
 
--- | Actual simplifier
-simplify :: M.ToBuiltinMeaning uni fun => Term TyName Name uni fun b -> Term TyName Name uni fun b
-simplify = DeadCode.removeDeadBindings . Inline.inline . Beta.beta
+-- | Run the simplifier @n@ times
+simplify' :: forall uni fun b. M.ToBuiltinMeaning uni fun => Integer -> Term TyName Name uni fun b -> Term TyName Name uni fun b
+simplify' n t | n <= 0    = t
+              | otherwise = simplify' (n - 1) (simplify t)
```

Comment:
Still not sure how I feel about this, but I recently found a "nice" way of doing this with Nikos.

```
appEndo (stimes n $ Endo f)
```
will apply f n times.

Reply:
Ah, thanks!